domino_data/auth.py

```diff
@@ -13,6 +13,7 @@
 
 
 @backoff.on_exception(backoff.expo, httpx.HTTPStatusError, max_time=2)
+@backoff.on_exception(backoff.expo, httpx.ReadTimeout, max_tries=2)
```

Comment:
This line is for mitigating the read timeout exception when getting the jwt token. 
All the other changes are for fixing the build failures caused by recent vulnerability.

Reply:
you should be able to use a tuple of exceptions instead of having two wrappers.

```python
@backoff.on_exception(backoff.expo, (httpx.HTTPStatusError, httpx.ReadTimeout), max_time=2)
def ...
```

Reply:
I thought about that idea too. The two exception fail in different pace.  HTTPStatusError fails fast, for ReadTimeout exception, it normally takes 5 seconds to fail.